core-bundle/src/Resources/contao/library/Contao/System.php

```diff
@@ -168,24 +169,17 @@ protected function import($strClass, $strKey=null, $blnForce=false)
 			{
 				$this->arrObjects[$strKey] = $container->get($strClass);
 			}
-			elseif (strpos($strClass, '.') !== false)
+			elseif ($container instanceof Container && isset($container->getRemovedIds()[$strClass]))
```

Comment:
I don't think the `instanceof Container` check is necessary. We are not checking this above and below, either.

Reply:
`getRemovedIds` is not part of the interface, it has only been in the class since Symfony 3.4

Reply:
Shouldn't we check `\in_array('getRemovedIds', get_class_methods($container), true)` then?

Reply:
No, we require 3.4 minimum.

Reply:
IMO the `instanceof` check is correct here. The existence of `getRemovedIds()` doesn’t guarantee that it works as we expect it, but the BC promise of Symfony should guarantee that the `Container` class works as expected.

Reply:
We're checking for the method existence because there can be differences in our dependencies. Not so in the case, the method must always be there if that class is implemented (which we actually know it is, but the interface does not guarantee that).